plugins/modules/crypto_info.py

```diff
@@ -0,0 +1,277 @@
+        - Theoretically this should be the case for version 0.5 and higher.
+      type: bool
+    has_dsa_sign:
+      description:
+        - Whether signing with DSA keys is supported.
+        - Theoretically this should be the case for version 1.5 and higher.
+      type: bool
+    has_rsa:
+      description:
+        - Whether RSA keys are supported.
+        - Theoretically this should be the case for version 0.5 and higher.
+      type: bool
+    has_rsa_sign:
+      description:
+        - Whether signing with RSA keys is supported.
+        - Theoretically this should be the case for version 1.4 and higher.
+      type: bool
+    has_x25519:
+      description:
+        - Whether X25519 keys are supported.
+        - Theoretically this should be the case for version 2.0 and higher, depending on the libssl version used.
+      type: bool
+    has_x25519_serialization:
+      description:
+        - Whether serialization of X25519 keys is supported.
+        - Theoretically this should be the case for version 2.5 and higher, depending on the libssl version used.
+      type: bool
+    has_x448:
+      description:
+        - Whether X448 keys are supported.
+        - Theoretically this should be the case for version 2.5 and higher, depending on the libssl version used.
+      type: bool
+'''
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible_collections.community.crypto.plugins.module_utils.crypto.basic import (
+    CRYPTOGRAPHY_HAS_EC,
+    CRYPTOGRAPHY_HAS_EC_SIGN,
+    CRYPTOGRAPHY_HAS_ED25519,
+    CRYPTOGRAPHY_HAS_ED25519_SIGN,
+    CRYPTOGRAPHY_HAS_ED448,
+    CRYPTOGRAPHY_HAS_ED448_SIGN,
+    CRYPTOGRAPHY_HAS_DSA,
+    CRYPTOGRAPHY_HAS_DSA_SIGN,
+    CRYPTOGRAPHY_HAS_RSA,
+    CRYPTOGRAPHY_HAS_RSA_SIGN,
+    CRYPTOGRAPHY_HAS_X25519,
+    CRYPTOGRAPHY_HAS_X25519_FULL,
+    CRYPTOGRAPHY_HAS_X448,
+    HAS_CRYPTOGRAPHY,
+)
+
+try:
+    import cryptography
+    from cryptography.exceptions import UnsupportedAlgorithm
+except ImportError:
+    UnsupportedAlgorithm = Exception
+    CRYPTOGRAPHY_VERSION = None
+    CRYPTOGRAPHY_IMP_ERR = traceback.format_exc()
+else:
+    CRYPTOGRAPHY_VERSION = cryptography.__version__
+    CRYPTOGRAPHY_IMP_ERR = None
+
+
+CURVES = [
+    ('secp224r1', 'SECP224R1'),
+    ('secp256k1', 'SECP256K1'),
+    ('secp256r1', 'SECP256R1'),
+    ('secp384r1', 'SECP384R1'),
+    ('secp521r1', 'SECP521R1'),
+    ('secp192r1', 'SECP192R1'),
+    ('sect163k1', 'SECT163K1'),
+    ('sect163r2', 'SECT163R2'),
+    ('sect233k1', 'SECT233K1'),
+    ('sect233r1', 'SECT233R1'),
+    ('sect283k1', 'SECT283K1'),
+    ('sect283r1', 'SECT283R1'),
+    ('sect409k1', 'SECT409K1'),
+    ('sect409r1', 'SECT409R1'),
+    ('sect571k1', 'SECT571K1'),
+    ('sect571r1', 'SECT571R1'),
+    ('brainpoolP256r1', 'BrainpoolP256R1'),
+    ('brainpoolP384r1', 'BrainpoolP384R1'),
+    ('brainpoolP512r1', 'BrainpoolP512R1'),
+]
+
+
+def add_crypto_information(module, result):
```

Comment:
`module` is unused in this function. I'd also prefer not mutating an input `dict`, but returning a `dict` result as an argument to `dict.update` e.g.:
```python
result = {}
result.update(add_crypto_information()) # Maybe s/add/get/
```
But that's just preference.

Reply:
My idea was to have a set of functions for cryptographic aspects with the same signature, and iterate over them to call them all. That's why there's the `module` parameter - there's no need for it, but other such functions will need it (f.ex. to find executables with `module.get_bin_path`).

About the input parameter: I'll change that.